### DISCUSSION TITLE
Problemas com encodeURI

### DISCUSSION THREAD

**Discussion by aliancasistemas** [frontendbr/forum#2288] [Perguntas e Respostas]
Boa tarde gurus.
Estou fazendo uma integração onde busco via GET informações sobre PDF hospedados na amazon AWS.
Mas, quando recebo o Json e tento abrir a URL passada no parâmetro me deparo com um erro "Acesso negado" na url gerada
Mas para minha surpresa se eu copiar a URI e colar na variável ele funciona.
Segue abaixo minha consulta:

data = {
    "Retorno": 1,
    "Mensagem": "",
    "URL": "https://documento-eletronico.s3.amazonaws.com/2022/05/04/02/21/39/24321e21-4d69-47ff-ba19-7706713aa6a8/oficio_362_2021%20-%20Clicksign.pdf?X-Amz-. blablabla",
    "campo": "",
    "TpRetorno": ""
}

response = $.parseJSON(data);
          
          if(response.Retorno==1)
            {
                if (response.URL != '')
                    {
                      const uri = response.URL.trim(); 
                     //Aqui que está o problema se eu recebo o parâmetro por aqui da erro no destino
                    //<Code>AccessDenied</Code>
                    //<Message>Access Denied</Message>
                   //Porém se eu copiar a URI da vairável e colcar : const uri = 'https://documento-eletronico.s3.amazonaws.com/2022/05/04/02/21/39/24321e21-4d69-47ff-ba19-7706713aa6a8 blablabla'  funciona de boa
              

                      const encoded = encodeURI(uri);

                      try {

                        window.open(decodeURI(encoded));
                      } catch (e) {
                        console.error(e);
                      }
                    }

Alguma sugestão?

**Comment by aliancasistemas** [comment]
Se eu fizer dessa forma, funciona (colocando a variável passada no parâmetro)
![image](https://user-images.githubusercontent.com/104798112/166815100-722aa76b-f90b-4ff9-8b79-116ac59ff67e.png)
Mas se eu fizer assim, não funciona
![image](https://user-images.githubusercontent.com/104798112/166815555-0e0e58f9-2d74-4926-90ea-d173bb737db4.png)

**Comment by felquis** [comment]
Olá @aliancasistemas tudo bem? Qual o seu nome?

Você pode tentar usar window.open sem encodeURI e decodeURI para ver se funciona.

```js
if (response.UrlDocumento) {
  try {
    window.open(response.UrlDocumento)
  } catch(e) {
    console.error(e)
  }
}
```

**Comment by aliancasistemas** [comment]
Opa.. Boa tarde irmão, André. 
Já tentei sim e sempre o mesmo erro
![image](https://user-images.githubusercontent.com/104798112/166982033-d71838c7-6788-438a-b240-93cd45479a85.png)
Uma coisa que reparei : 
![image](https://user-images.githubusercontent.com/104798112/166982140-30067ef7-a8cf-4b75-bb9c-f4aebb5ee8a0.png)
Se eu coloco diretamente a URI na string "const url = "blablablabla", a função encodeURI e decodeURI funciona, porém se coloco recebendo o parâmetro "const url = response.URL" ele não funciona, mesmo se eu utilizar um url.replace('\u0026','&') também não funciona e gera a URL 
![image](https://user-images.githubusercontent.com/104798112/166983010-516bce48-2be5-4391-b6b6-56a74f1f154d.png)
Mas colando diretamente o valor passado no parâmetro ele funciona e gera a URL 
![image](https://user-images.githubusercontent.com/104798112/166983328-da456460-375a-4ac6-9375-204b6e48c155.png)
